cmd/snap-repair/runner.go

```diff
@@ -43,6 +47,138 @@ import (
 	"github.com/snapcore/snapd/strutil"
 )
 
+// Repair is a runnable repair.
+type Repair struct {
+	*asserts.Repair
+
+	run      *Runner
+	sequence int
+}
+
+func (r *Repair) RunDir() string {
+	return filepath.Join(dirs.SnapRepairRunDir, r.BrandID(), r.RepairID())
+}
+
+// SetStatus sets the status of the repair in the state and saves the latter.
+func (r *Repair) SetStatus(status RepairStatus) {
+	brandID := r.BrandID()
+	cur := *r.run.state.Sequences[brandID][r.sequence-1]
+	cur.Status = status
+	r.run.setRepairState(brandID, cur)
+	r.run.SaveState()
+}
+
+// nextCounter finds the next counter prefix to use for the output and status
+// files of a repair run
+func nextCounter(r *Repair) int {
+	counter := 0
+
+	// format: "r99.000001.2006-01-02T150405.999999999.output"
```

Comment:
Indeed, sorry. Now that we have the counter we don't need the nano precision. Before with low precision file would get clobbered.